### DISCUSSION TITLE
CloudWatch get times to invoke from DynamoDB and invoke Lambda Function

### DISCUSSION THREAD

**Discussion by MedNabilEssefaihi** [aws/aws-sdk-js#4087] [General]
Dear Support Team,

I am trying to make **automatic functions** by **CloudWatch**.
It has to work by specific times exist in **DynamoDB** documents.

### My Need
I do have fixed times (3 specific times) for each user document in DynamoDB.
My need is of call a specific function in Lambda Function whenever the time is come (it's like an alarm system)

Users Dcouments Example:
```
{
  {
        "id": 1xxxxxxxxxxxxxxxxx,
        "tasksStart":{
         "morning": "07:00",
         "afternoon":"14:00",
         "evening":"19:00",
        }
  },
  {
        "id": 2xxxxxxxxxxxxxxxxx,
        "tasksStart":{
         "morning": "08:30",
         "afternoon":"12:10",
         "evening":"20:00",
        }
  },
  ...
}
```
### What I did
I made a Lambda Function has a 3 functions to work
1. goodMorning()
2. goodAfternoon()
3. goodEvening()

```
const resolvers = {
  Mutation: {
    goodMorning: async (ctx) => {
       return await morningToDos(ctx.arguments.id);
    },
    goodAfternoon: async (ctx) => {
      return await journeyToDos(ctx.arguments.id);
    },
    goodEvening: async (ctx) => {
     return await eveningToDos(ctx.arguments.id);
    },
  },
};

exports.handler = async (event) => {
  const typeHandler = resolvers[event.typeName];

  if (typeHandler) {
    const resolver = typeHandler[event.fieldName];
    if (resolver) {
      const result = await resolver(event);
      return result;
    } else {
      return null;
    }
  }
};

```
Example:
- for user 1, if it is **07:00** in the morning, lambda function needs to invoke function called **goodMorning**.
- for user 2, if it is **08h30** in the morning,  lambda function needs to invoke function called **goodMorning**.
etc...

### what should do ?

**Comment by ajredniwja** [comment]
Hi @MedNabilEssefaihi thanks for opening this, I think its more like how you can design the code but not the SDK. If I understand your question correctly you want to invoke a lambda according to clients time or timezones. I found this stackOverflow discussion that might be helpful in your case if I understand the question correctly: https://stackoverflow.com/questions/10659523/how-to-get-the-exact-local-time-of-client

Then you can call the specific function depending upon what time is returned.

**Comment by MedNabilEssefaihi** [comment]
Hello @ajredniwja, 

Thank you so much for time and your effort. Actually, my need is to make an automatic system does work independent. I mean the serverless (Lambda Function) knows the times and it works automatically. without making any code from the front-end

**Comment by ajredniwja** [comment]
@MedNabilEssefaihi sorry I lost track of this issue, is it still a persisting issue?
